### PR TITLE
SplitMode for SPM Package

### PR DESCRIPTION
**Pull request by mackoj**
This is another try at https://github.com/yonaskolb/SwagGen/pull/225.

## TL;DR

This **change nothing** to all actual users of Swaggen. It add the support of splitting the generated code for SPM user only and only if they want to. This is the first step to remove Alamofire. It allow to run generated code on Linux if you only use the request and model targets. And finally it add dynamic library support when using SPM.

## What have been done

This PR is focus to mainly focus to split in SPM the generated code into 4 part:
- Client: take an URLRequest and send them
- Models: all the generated models and enums
- Requests: all the generated request builders - this part generate URLRequest
- SharedCode: Shared code between various targets

```
├── Client
│   ├── APIClient.swift
│   └── RequestBehaviour.swift
├── Models
│   ├── APIModel.swift
│   ├── AnyCodable.swift
│   ├── Coding.swift
│   ├── Enum.swift
│   └── Model.swift
├── Requests
│   ├── API.swift
│   ├── APIRequest.swift
│   ├── APIResponse.swift
│   ├── APIService.swift
│   ├── AnyRequest.swift
│   └── Request.swift
└── SharedCode
    ├── APIClientError.swift
    └── APIResult.swift
    └── Timeline.swift
```

## Why

I did these changes because:
- I needed to have a generated library working on Linux and Alamofire even the version 5 don't support Linux(https://github.com/Alamofire/Alamofire/pull/3098)
- Simplify the PR that will move Alamofire out
- I needed to onboard new developper to SwagGen development
- I needed the dynamic library support for a big project that I work on that use XcodeGen and some module are imported in multiple place so we needed the [dynamic library linking](https://github.com/yonaskolb/XcodeGen/pull/788) in order to making it work

## Remarks

I don't think this should be another template because it doesn’t change the main template it make it more flexible.

## Point of interest to the reviewer

In order to make everything works I have to make `createURLRequest` but it worked with Alamofire so I needed to import the relevant part from Alamofire to make it work.
I did copy the Alamofire Timeline file in order to keep the feature.

## Todo after review (in order to not clutter the main work of this PR)

- we should add support for CI for Linux after this is merged.
- adding test on podspec in CI could be great too.
- updating fixtures

I hope it helps.

### CODE REVIEW

**Comment by mackoj** [comment]
Update from #230 modifications.

**Comment by mackoj** [comment]
Rebase for #239
